ProfessorAssessor/View Controllers/LogInViewController.m

```diff
@@ -57,26 +57,38 @@ - (void)displayFacebookLoginWithManager:(FBSDKLoginManager *)manager
 }
 
 - (void)fetchUserInformationWithRequest:(FBSDKGraphRequest *)request {
-    [request startWithCompletion:^(id<FBSDKGraphRequestConnecting> _Nullable connection, id _Nullable result, NSError *_Nullable error) {
+    [request
+     startWithCompletion:^(
+                           id<FBSDKGraphRequestConnecting> _Nullable connection,
+                           id _Nullable result,
+                           NSError *_Nullable error) {
         if (result) {
-            self.requestResult = @{@"first_name":result[@"first_name"], @"last_name":result[@"last_name"], @"email":result[@"email"]};
-            NSString *email = result[@"email"];
+            FacebookUser *user = [FacebookUser
+                                  createUserWithFirstName:result[@"first_name"]
+                                  lastName:result[@"last_name"]
+                                  email:result[@"email"]];
 
-            [self handleFacebookLoginWithEmail:email];
+            [self completeLoginWithFacebookUser:user];
         }
     }];
 }
 
-- (void)handleFacebookLoginWithEmail:(NSString *)email {
+- (void)completeLoginWithFacebookUser:(FacebookUser *)user {
     PFQuery *userQuery = [User query];
 
-    [userQuery whereKey:@"email" equalTo:email];
+    [userQuery whereKey:@"email" equalTo:user.email];
 
-    [userQuery countObjectsInBackgroundWithBlock:^(int count, NSError *_Nullable error) {
+    [userQuery
+     countObjectsInBackgroundWithBlock:^(
+                                         int count,
+                                         NSError *_Nullable error) {
         if (count == 0) {
-            [self performSegueWithIdentifier:@"signUpSegue" sender:self];
+            [self presentAlertWithTitle:nil withMessage:@"No account associated with this Facebook login. Please click 'Back' to sign up."];
```

Comment:
I'll check if you did this in the next pr but you should be able to just sign them up at this point.